locator/network_topology_strategy.cc

```diff
@@ -315,16 +315,13 @@ static future<unsigned> calculate_initial_tablets_from_topology(const schema& s,
             shards_per_dc_map[n.dc_rack().dc] += n.get_shard_count();
         }
     });
+    min_per_shard_tablet_count = std::max(1.0, min_per_shard_tablet_count);
     for (const auto& [dc, rf_in_dc] : rf) {
         if (!rf_in_dc) {
             continue;
         }
         unsigned shards_in_dc = shards_per_dc_map[dc];
-        unsigned tablets_in_dc = (shards_in_dc + rf_in_dc - 1) / rf_in_dc;
-        if (min_per_shard_tablet_count) {
-            auto min_tablets_in_dc = std::ceil((double)(min_per_shard_tablet_count * shards_in_dc) / rf_in_dc);
-            tablets_in_dc = std::max<unsigned>(tablets_in_dc, min_tablets_in_dc);
-        }
+        unsigned tablets_in_dc = std::ceil((double)(min_per_shard_tablet_count * shards_in_dc) / rf_in_dc);
```

Comment:
I understand it's just a simplification, but I would like to allow values 0 < min_per_shard_tablet_count < 1 one day. For example 0.5 would mean that we target half the shards in the cluster to have a tablet. But not worth dwelling on now.

Reply:
Agreed, in large clusters serving a large number of small tables, shards will have to serve a subset of the tables to satisfy the global per-shard tablet count goal.

Reply:
I agree too.

Reply:
The question is whether we need to keep the current logic.
I think that if the `min_per_shard_tablet_count` option is specified and is less than one we should respect it.
If it isn't specified than the previous patch will use the initial scale factor or 1 which satisfies the max expression above.

Reply:
In V3 I changed the code to respect min_per_shard_tablet_count / initial_scale smaller than 1.